### PR TITLE
apache2: add LWRP for WSGI services

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

**Comment by dirkmueller** [comment]
given that the apache2 cookbook is supposed to be synced with upstream again at some point, how about we move that to the openstack cookbook instead?

**Comment by aplanas** [comment]
@dirkmueller ok for me

File: chef/cookbooks/apache2/providers/wsgi.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          processes: current_resource.processes,
+          threads: current_resource.threads,
+          process_group: current_resource.process_group,
+          script_alias: current_resource.script_alias,
+          directory: current_resource.directory,
+          access_log: current_resource.access_log,
+          error_log: current_resource.error_log,
+          apache_log_dir: node[:apache][:log_dir],
+        )
+        notifies :restart, resources(service: "apache2"), :immediately
```
Comment (vuntz) [inline chef/cookbooks/apache2/providers/wsgi.rb]: s/restart/reload/?

File: chef/cookbooks/apache2/providers/wsgi.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    end
+  end
+end
+
+action :delete do
+  if @current_resource.exists
+    converge_by("Delete #{@new_resource}") do
+      file _get_vhost_name do
+        action :delete
+        only_if { File.exist?(_get_vhost_name) }
```
Comment (vuntz) [inline chef/cookbooks/apache2/providers/wsgi.rb]: Missing notification for apache?

Comment (aplanas) [inline chef/cookbooks/apache2/providers/wsgi.rb]: Done, thanks!

File: chef/cookbooks/apache2/providers/wsgi.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
+# Support whyrun
```
Comment (toabctl) [inline chef/cookbooks/apache2/providers/wsgi.rb]: What about a license header?

File: chef/cookbooks/apache2/providers/wsgi.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  end
+
+  @current_resource
+end
+
+private
+
+def _get_vhost_name
+  apache_dir = node[:apache][:dir]
+  "#{apache_dir}/vhosts.d/#{@new_resource.daemon_process}.conf"
```
Comment (toabctl) [inline chef/cookbooks/apache2/providers/wsgi.rb]: vhosts.d is suse specific. But I guess that is ok for now?!

File: chef/cookbooks/apache2/providers/wsgi.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+def _get_group
+  @new_resource.group || @new_resource.user
+end
+
+def _get_process_group
+  @new_resource.process_group || @new_resource.daemon_process
+end
+
+def _get_script_alias
+  default_script_alias = "/srv/www/#{@new_resource.daemon_process}/app.wsgi"
```
Comment (toabctl) [inline chef/cookbooks/apache2/providers/wsgi.rb]: same here

**Comment by toabctl** [comment]
@aplanas do you already have a PR which uses this new LWRP? Maybe you can just add a commit here and convert one of our apache consumers (eg keystone)?

**Comment by aplanas** [comment]
@toabctl moved the code to crowbar-openstack cookbook: https://github.com/crowbar/crowbar-openstack/pull/637

There is also a use for barbican

File: chef/cookbooks/apache2/providers/wsgi.rb (hunk 4 above)
Comment (aplanas) [inline chef/cookbooks/apache2/providers/wsgi.rb]: IMHO is ok for now ..

**Pull request closed without merging**